### PR TITLE
core/linux-aarch64 to 6.6.12-1

### PR DESCRIPTION
**Pull request by cbix**
Successfully tested on Raspberry Pi Zero 2
For testing you may use binaries attached [here](https://github.com/cbix/PKGBUILDs/releases/tag/linux-aarch64-6.6.12-1) (`pacman -U ...pkg.tar.xz`)

### CODE REVIEW

**Comment by cbix** [comment]
Ping @kmihelich

**Comment by UniqueUsername5** [comment]
I can also confirm that it works on my Odroid C4.
There is, however, a conflict between linux-aarch64 and linux-aarch64-headers because of the symlink usr/lib/modules/6.6.1-1-aarch64-ARCH/build that is created in the linux-aarch64 package.

I have added the necessary rm command from the mainline archlinux kernel PKGBUILD to yours and attached it to this comment.
I also bumped the version to 6.6.1

[PKGBUILD.tar.gz](https://github.com/archlinuxarm/PKGBUILDs/files/13327889/PKGBUILD.tar.gz)

**Comment by graysky2** [comment]
Probably better if all needed mods are incorporated into the PR...

**Comment by chriswyatt** [comment]
Why the force push?

**Comment by cbix** [comment]
> Why the force push?

Just to rebase, update and add the fix from @UniqueUsername5. But since upstream can just squash & merge it really doesn't matter :)

**Comment by chriswyatt** [comment]
It's not me reviewing, so meh

**Comment by chriswyatt** [comment]
Nevermind

**Comment by cbix** [comment]
Updated to enable the config from this PR
- #2010

**Comment by lategoodbye** [comment]
I strongly suggest to update to 6.6.2, which contains a bugfix for the bcm2835 hardware random generator driver:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.2&id=d0f3994c6a40bcdcf8a2e3ce9b4653e323a44335

File: core/linux-aarch64/PKGBUILD [hunk 1]
```diff
@@ -41,16 +41,15 @@ prepare() {
   cd $_srcname
 
   echo "Setting version..."
-  scripts/setlocalversion --save-scmversion
   echo "-$pkgrel" > localversion.10-pkgrel
   echo "${pkgbase#linux}" > localversion.20-pkgname
 
   # add upstream patch
-  git apply --whitespace=nowarn ../patch-${pkgver}
+  patch -Np1 < ../patch-${pkgver}
```
Comment (cbix) [inline core/linux-aarch64/PKGBUILD:48]: `git apply` didn't do anything here for some reason

Comment (anarsoul) [reply core/linux-aarch64/PKGBUILD:48]: it doesn't work if you build linux-aarch64 inside of PKGBUILDs git tree

Comment (cbix) [reply core/linux-aarch64/PKGBUILD:48]: I see, so the package maintainers are just building using an external `SRCDEST` by default?

**Comment by nvidialinuxuser** [comment]
Nice work, you all!

@cbix, I'm not exactly sure why, but for some reason the ALARM devs don't seen to want to update certain packages.  I doubt that this PR is going go through anytime soon (if at all) rather unfortunately.

In fact, this is not the only ALARM component that needs updating.  There's several broken/outdated packages that need to be taken care of.  The situation is worse for armv7h.  Perhaps it's time to end armv7h support?

One instance of packages not being updated is Glibc (for both armv7h and aarch64) in the ALARM repos.  It is stuck at 2.35 but its respective PKGBUILD is being updated it appears on both aarch64/armv7h.  The latest release of every major Linux distro (e.g. Ubuntu, Debian, Fedora, etc) has a newer Glibc.  In fact, until there isn't an updated Glibc in the repos, this very upstream-friendly mainline WideVine solution can't be used at all: [https://github.com/AsahiLinux/widevine-installer](https://github.com/AsahiLinux/widevine-installer)

I would recommend adding these patches though for 'Gru' model ChromeBooks as it'll improve their mainline support:

[cadmium-gru.0001-drm-bridge-analogix_dp-Don-t-return-EBUSY-when-msg-s.patch](https://github.com/lynix/linux-aarch64-lts/blob/master/cadmium-gru.0001-drm-bridge-analogix_dp-Don-t-return-EBUSY-when-msg-s.patch)
[cadmium-gru.0002-drm-rockchip-Only-wait-for-panel-ACK-on-PSR-entry.patch](https://github.com/lynix/linux-aarch64-lts/blob/master/cadmium-gru.0002-drm-rockchip-Only-wait-for-panel-ACK-on-PSR-entry.patch)

I had those patches sent in for a well-maintained unofficial linux-aarch64-lts kernel for ALARM.  Though said project doesn't support ARM64 ChromeBooks officially (as ARM64 ChromeBooks support DepthCharge by default), I sent those patches in because I have a 'Gru' ChromeBook that uses LibreBoot using the U-Boot payload instead of DepthCharge.

Hope you all have a nice evening!

**Comment by cbix** [comment]
> don't seen to want to update certain packages

Not sure if they don't _want_ to, it certainly is a lot of work for the small number of maintainers. I really hope Arch Linux will eventually unite with ALARM and/or at least adopt aarch64 officially like most major distributions out there.

For now I'll just wait, if I ever urgently need some packages updated in a repo I'll just create my own repo for that.

**Comment by anarsoul** [comment]
Can someone merge it please? @kmihelich

linux-aarch64 hasn't been updated since April 2023 and 6.2.10 is not supported anymore. If the maintainers do not have time to review or test linux-aarch64 maybe it's time to expand the team? I volunteer to support linux-aarch64 package if you need more working hands.

**Comment by nvidialinuxuser** [comment]
> Can someone merge it please? @kmihelich
> 
> linux-aarch64 hasn't been updated since April 2023 and 6.2.10 is not supported anymore. If the maintainers do not have time to review or test linux-aarch64 maybe it's time to expand the team? I volunteer to support linux-aarch64 package if you need more working hands.

Someone is actively working on an LTS version of the Linux package for ALARM, but it has no DepthCharge support for ChromeBooks that do not run on LibreBoot.

[https://github.com/lynix/linux-aarch64-lts](https://github.com/lynix/linux-aarch64-ltsl)

There was this discussion that I participated in: [https://bbs.archlinux.org/viewtopic.php?id=290931](https://bbs.archlinux.org/viewtopic.php?id=290931l)

Maybe regular Arch Linux will start supporting ARM64 someday.

**Comment by Danct12** [comment]
> maybe it's time to expand the team?

I'm happy to give them a hand.

**Comment by graysky2** [comment]
I updated my own fork to 6.7.4 here: https://github.com/graysky2/PKGBUILDs/commit/7bd695c7cc23cbb4e62250720af05ead68776d2d

Can you guys build/test on your hardware?  Works for me with RPi4B if I boot with the updated uboot-raspberrypi (just pushed to master) and keeping raspberrypi-bootloadered held at version 20240122-1 due to this bug: https://github.com/raspberrypi/firmware/issues/1868

**Comment by cbix** [comment]
@graysky2 just started building, first thing I noticed is it's prompting for some new kconfig entries:
- `MFD_TI_AM335X_TSCADC`
- `MMC_SDHCI_OMAP`
- `MMC_SDHCI_AM654`
Ideally the whole generated config should be present so the build can run unattended :)
Will report later if it runs on my Zero 2W.

**Comment by graysky2** [comment]
That is odd... I did not get any prompts when I built it.  Reviewing.

**Comment by graysky2** [comment]
That is odd... I did not get any prompts when I built it. Reviewing.

EDIT: no such issue:
```
% makepkg -s
==> Making package: linux-aarch64 6.7.4-1 (Thu 15 Feb 2024 12:59:28 PM EST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Installing missing dependencies...
resolving dependencies...
looking for conflicting packages...

Packages (6) docbook-xml-4.5-9  docbook-xsl-1.79.2-7  inetutils-2.5-1  uboot-tools-2024.01-2  vboot-utils-r2729.10343b821-1
             xmlto-0.0.28-4

Total Download Size:    1.82 MiB
Total Installed Size:  31.21 MiB

:: Proceed with installation? [Y/n] 
:: Retrieving packages...
 xmlto-0.0.28-4-aarch64                           25.0 KiB   125 KiB/s 00:00 [############################################] 100%
 docbook-xml-4.5-9-any                            84.0 KiB   304 KiB/s 00:00 [############################################] 100%
 vboot-utils-r2729.10343b821-1-aarch64           196.1 KiB   613 KiB/s 00:00 [############################################] 100%
 uboot-tools-2024.01-2-aarch64                   185.8 KiB   547 KiB/s 00:00 [############################################] 100%
 docbook-xsl-1.79.2-7-any                       1374.0 KiB  1789 KiB/s 00:01 [############################################] 100%
 Total (5/5)                                    1865.0 KiB  2.25 MiB/s 00:01 [############################################] 100%
(6/6) checking keys in keyring                                               [############################################] 100%
(6/6) checking package integrity                                             [############################################] 100%
(6/6) loading package files                                                  [############################################] 100%
(6/6) checking for file conflicts                                            [############################################] 100%
:: Processing package changes...
(1/6) installing xmlto                                                       [############################################] 100%
(2/6) installing docbook-xml                                                 [############################################] 100%
(3/6) installing docbook-xsl                                                 [############################################] 100%
(4/6) installing inetutils                                                   [############################################] 100%
(5/6) installing uboot-tools                                                 [############################################] 100%
(6/6) installing vboot-utils                                                 [############################################] 100%
:: Running post-transaction hooks...
(1/4) Creating system user accounts...
Creating group '_talkd' with GID 960.
Creating user '_talkd' (User for legacy talkd server) with UID 960 and GID 960.
(2/4) Reloading system manager configuration...
(3/4) Arming ConditionNeedsUpdate...
(4/4) Updating the info directory file...
==> Retrieving sources...
  -> Downloading linux-6.7.tar.xz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   162  100   162    0     0   1340      0 --:--:-- --:--:-- --:--:--  1350
100   162  100   162    0     0    526      0 --:--:-- --:--:-- --:--:--   526
100  134M  100  134M    0     0  3187k      0  0:00:43  0:00:43 --:--:-- 5082k
  -> Downloading patch-6.7.4.xz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   162  100   162    0     0   1737      0 --:--:-- --:--:-- --:--:--  1741
100   162  100   162    0     0    605      0 --:--:-- --:--:-- --:--:--   605
100  437k  100  437k    0     0   662k      0 --:--:-- --:--:-- --:--:-- 2254k
  -> Found 0001-net-smsc95xx-Allow-mac-address-to-be-set-as-a-parame.patch
  -> Found 0002-arm64-dts-rockchip-disable-pwm0-on-rk3399-firefly.patch
  -> Found config
  -> Found generate_chromebook_its.sh
  -> Found kernel.keyblock
  -> Found kernel_data_key.vbprivk
  -> Found linux.preset
==> Validating source files with md5sums...
    linux-6.7.tar.xz ... Passed
    patch-6.7.4.xz ... Passed
    0001-net-smsc95xx-Allow-mac-address-to-be-set-as-a-parame.patch ... Passed
    0002-arm64-dts-rockchip-disable-pwm0-on-rk3399-firefly.patch ... Passed
    config ... Passed
    generate_chromebook_its.sh ... Passed
    kernel.keyblock ... Passed
    kernel_data_key.vbprivk ... Passed
    linux.preset ... Passed
==> Extracting sources...
  -> Extracting linux-6.7.tar.xz with bsdtar
  -> Extracting patch-6.7.4.xz with xz
==> Starting prepare()...
Setting version...
==> Starting build()...
  SYNC    include/config/auto.conf
  HOSTCC  scripts/basic/fixdep
  HOSTCC  scripts/kconfig/conf.o
  HOSTCC  scripts/kconfig/confdata.o
  HOSTCC  scripts/kconfig/expr.o
...
```

**Comment by cbix** [comment]
@graysky2 sorry, my mistake, since on your branch the PKGBUILD is using `git apply` for patching I needed to build outside the git working directory, otherwise it would just build 6.7.0 without any patches.

**Comment by anarsoul** [comment]
@graysky2 It doesn't boot on rock64 (and likely won't boot on other rockchip devices). It lacks MFD_RK8XX_I2C in kernel config, so RK808 driver is not built.

**Comment by anarsoul** [comment]
@graysky2 this should do the trick: https://gist.github.com/anarsoul/e3a12bcdd165618471084b51c2711249

I'm rebuilding the package atm, it'll take couple hours. I'll test it tomorrow on rock64, rockpro64 and rock pi 4. Another thing I noticed is that installing the package didn't trigger rebuilding initramfs for some reason.

**Comment by graysky2** [comment]
I pulled in your gist and updated to 6.7.5.

**Comment by graysky2** [comment]
> I'm rebuilding the package atm, it'll take couple hours. I'll test it tomorrow on rock64, rockpro64 and rock pi 4
I pulled your config changes and bumped to 6.7.5.

> Another thing I noticed is that installing the package didn't trigger rebuilding initramfs for some reason.

That is odd, for me it is consistently triggered:
```
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) linux-aarch64-6.7.5-1

Total Installed Size:  158.10 MiB
Net Upgrade Size:        0.04 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                               [############################################] 100%
(1/1) checking package integrity                                             [############################################] 100%
(1/1) loading package files                                                  [############################################] 100%
(1/1) checking for file conflicts                                            [############################################] 100%
:: Processing package changes...
(1/1) upgrading linux-aarch64                                                [############################################] 100%
:: Running post-transaction hooks...
(1/3) Arming ConditionNeedsUpdate...
(2/3) Updating module dependencies...
(3/3) Updating linux initcpios...
==> Building image from preset: /etc/mkinitcpio.d/linux-aarch64.preset: 'default'
==> Using configuration file: '/etc/mkinitcpio.conf'
  -> -k 6.7.5-1-aarch64-ARCH -c /etc/mkinitcpio.conf -g /boot/initramfs-linux.img
==> Starting build: '6.7.5-1-aarch64-ARCH'
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [autodetect]
  -> Running build hook: [modconf]
  -> Running build hook: [kms]
  -> Running build hook: [keyboard]
  -> Running build hook: [keymap]
  -> Running build hook: [block]
  -> Running build hook: [filesystems]
  -> Running build hook: [fsck]
==> Generating module dependencies
==> Creating gzip-compressed initcpio image: '/boot/initramfs-linux.img'
==> WARNING: errors were encountered during the build. The image may not be complete.
==> Building image from preset: /etc/mkinitcpio.d/linux-aarch64.preset: 'fallback'
==> Using configuration file: '/etc/mkinitcpio.conf'
  -> -k 6.7.5-1-aarch64-ARCH -c /etc/mkinitcpio.conf -g /boot/initramfs-linux-fallback.img -S autodetect
==> Starting build: '6.7.5-1-aarch64-ARCH'
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [modconf]
  -> Running build hook: [kms]
==> WARNING: Possibly missing firmware for module: 'ast'
==> WARNING: Possibly missing firmware for module: 'msm'
  -> Running build hook: [keyboard]
  -> Running build hook: [keymap]
  -> Running build hook: [block]
==> WARNING: Possibly missing firmware for module: 'qla2xxx'
==> WARNING: Possibly missing firmware for module: 'qla1280'
==> WARNING: Possibly missing firmware for module: 'wd719x'
  -> Running build hook: [filesystems]
  -> Running build hook: [fsck]
==> Generating module dependencies
==> Creating gzip-compressed initcpio image: '/boot/initramfs-linux-fallback.img'
==> WARNING: errors were encountered during the build. The image may not be complete.
error: command failed to execute correctly
```

**Comment by anarsoul** [comment]
@graysky2 did you try updating from 6.2.10?

**Comment by graysky2** [comment]
Yes.
```
% uname -a
Linux rpi4 6.2.10-1-aarch64-ARCH #1 SMP PREEMPT_DYNAMIC Fri Apr  7 10:32:52 MDT 2023 aarch64 GNU/Linux
# pacman -U linux-aarch64-6.7.5-1-aarch64.pkg.tar.zst
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) linux-aarch64-6.7.5-1

Total Installed Size:  158.10 MiB
Net Upgrade Size:       14.22 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                               [############################################] 100%
(1/1) checking package integrity                                             [############################################] 100%
(1/1) loading package files                                                  [############################################] 100%
(1/1) checking for file conflicts                                            [############################################] 100%
:: Processing package changes...
(1/1) upgrading linux-aarch64                                                [############################################] 100%
:: Running post-transaction hooks...
(1/3) Arming ConditionNeedsUpdate...
(2/3) Updating module dependencies...
(3/3) Updating linux initcpios...
==> Building image from preset: /etc/mkinitcpio.d/linux-aarch64.preset: 'default'
==> Using configuration file: '/etc/mkinitcpio.conf'
  -> -k 6.7.5-1-aarch64-ARCH -c /etc/mkinitcpio.conf -g /boot/initramfs-linux.img
==> Starting build: '6.7.5-1-aarch64-ARCH'
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [autodetect]
  -> Running build hook: [modconf]
  -> Running build hook: [kms]
  -> Running build hook: [keyboard]
  -> Running build hook: [keymap]
  -> Running build hook: [block]
  -> Running build hook: [filesystems]
  -> Running build hook: [fsck]
==> Generating module dependencies
==> Creating gzip-compressed initcpio image: '/boot/initramfs-linux.img'
==> WARNING: errors were encountered during the build. The image may not be complete.
==> Building image from preset: /etc/mkinitcpio.d/linux-aarch64.preset: 'fallback'
==> Using configuration file: '/etc/mkinitcpio.conf'
  -> -k 6.7.5-1-aarch64-ARCH -c /etc/mkinitcpio.conf -g /boot/initramfs-linux-fallback.img -S autodetect
==> Starting build: '6.7.5-1-aarch64-ARCH'
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [modconf]
  -> Running build hook: [kms]
==> WARNING: Possibly missing firmware for module: 'ast'
==> WARNING: Possibly missing firmware for module: 'msm'

  -> Running build hook: [keyboard]
  -> Running build hook: [keymap]
  -> Running build hook: [block]
==> WARNING: Possibly missing firmware for module: 'qla2xxx'
==> WARNING: Possibly missing firmware for module: 'qla1280'
==> WARNING: Possibly missing firmware for module: 'wd719x'
  -> Running build hook: [filesystems]
  -> Running build hook: [fsck]
==> Generating module dependencies
==> Creating gzip-compressed initcpio image: '/boot/initramfs-linux-fallback.img'
==> WARNING: errors were encountered during the build. The image may not be complete.
error: command failed to execute correctly
```

**Comment by graysky2** [comment]
@anarsoul - you can grab a version I compiled from [here](http://repo-ck.com/linux-aarch64-test/)..  Does your rock64 boot and run as it should?

**Comment by anarsoul** [comment]
> Does your rock64 boot and run as it should? Also, if you're planning to test multiple SoCs, please list them.

I already built my own version from your PKGBUILD. It boots fine on rock64 (rk3328) and rock pi 4b (rk3399). I'll also test it on rockpro64 (another rk3399) later today. If you need to test it on more SoCs I can try booting it on Pine64 (Allwinner A64)

**Comment by graysky2** [comment]
No, I just pushed.  Thanks.  This PR is closed via https://github.com/archlinuxarm/PKGBUILDs/commit/0f2bbe5123fa5c2dc11650794250450a756f71ca thanks all.

**Comment by anarsoul** [comment]
> No, I just pushed. Thanks. This PR is closed via [0f2bbe5](https://github.com/archlinuxarm/PKGBUILDs/commit/0f2bbe5123fa5c2dc11650794250450a756f71ca) thanks all.

Thanks a lot for taking care of this!

**Comment by anarsoul** [comment]
Just for the record, it boots fine on Rockpro64 and Pine64-LTS

**Pull request closed without merging**